addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobStoreConfiguration.java

```diff
@@ -521,6 +552,9 @@ protected TransferManager createTransferManager() {
                                              MULTIPART_COPY_THRESHOLD_PROPERTY, MULTIPART_COPY_THRESHOLD_DEFAULT)))
                                      .withMultipartCopyPartSize(Long.valueOf(getMultipartCopyPartSize()))
                                      .withAlwaysCalculateMultipartMd5(alwaysCalculateMultipartMd5)
+                                     .withExecutorFactory(() -> Executors.newFixedThreadPool(Integer.valueOf(
+                                             getIntProperty(TRANSFER_MANAGER_THREAD_POOL_SIZE_PROPERTY,
+                                                     TRANSFER_MANAGER_THREAD_POOL_SIZE_DEFAULT))))
```

Comment:
`Executors.newFixedThreadPool((int) getLongProperty(...))` is enough.

Note however that the AWS default is `TransferManagerUtils.createDefaultExecutorService()` which does 

        ThreadFactory threadFactory = new ThreadFactory() {
            private int threadCount = 1;
            public Thread newThread(Runnable r) {
                Thread thread = new Thread(r);
                thread.setName("s3-transfer-manager-worker-" + threadCount++);
                return thread;
            }
        };
        return (ThreadPoolExecutor) Executors.newFixedThreadPool(10, threadFactory);

So we're losing the naming of the threads with the proposed solution.

Reply:
Agreed with this solution as it helps us to diagnose issues in production when it comes to identifying threads.